src/react/ReactQueryConfigProvider.tsx

```diff
@@ -1,19 +1,30 @@
 import React from 'react'
-import { DEFAULT_CONFIG, defaultConfigRef } from '../core/config'
 
-//
+import { DEFAULT_CONFIG, defaultConfigRef } from '../core/config'
+import { ReactQueryConfig } from '../core/types'
 
-const configContext = React.createContext()
+const configContext = React.createContext<ReactQueryConfig | undefined>(
+  undefined
+)
 
 export function useConfigContext() {
   return React.useContext(configContext) || defaultConfigRef.current
 }
 
-export function ReactQueryConfigProvider({ config, children }) {
-  let configContextValueOrDefault = useConfigContext()
-  let configContextValue = React.useContext(configContext)
+export interface ReactQueryProviderConfig extends ReactQueryConfig {}
```

Comment:
Why is this needed? We should ideally be using `ReactQueryConfig` itself at other places right?

Reply:
I mostly extended here to keep the exported type name the same as the current one. The provider could accept different options in the future though, but I don't think it is likely